core/src/main/scala/org/scalafmt/rewrite/AvoidInfix.scala

```diff
@@ -0,0 +1,53 @@
+package org.scalafmt.rewrite
+
+import org.scalafmt.util.TokenOps
+
+import scala.meta.tokens.Token.{LF, LeftBrace, LeftParen}
+import scala.meta.{Tree, _}
+
+case object AvoidInfix extends Rewrite {
+  // In a perfect world, we could just use
+  // Tree.transform {
+  //   case t: Term.ApplyInfix => Term.Apply(Term.Select(t.lhs, t.op), t.args)
+  // }
+  // and be done with it. However, until transform becomes token aware (see https://github.com/scalameta/scalameta/pull/457)
+  // we will do these dangerous rewritings by hand.
+
+  override def rewrite(code: Tree, ctx: RewriteCtx): Seq[Patch] = {
+    val matcher = ctx.style.rewrite.neverInfix.toMatcher
+    code.collect {
+      case Term.ApplyInfix(lhs, op, targs, args)
+          if matcher.matches(op.value) =>
+        val fstOpToken = op.tokens.head
+
+        val selectorToBeAdded = Seq(
+          TokenPatch.Add(fstOpToken, ".", "", true)
```

Comment:
It is like this because of true (keepTok), by default i set it to false. False makes more sense for all rewrites except AvoidInfix. But i can introduce Token.AddLeft(token, keepTok)